src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -384,6 +377,19 @@ RouterExecutorRun(QueryDesc *queryDesc, ScanDirection direction, long count)
 		(*destination->rStartup)(destination, operation, queryDesc->tupDesc);
 	}
 
+	/* we only support returning nothing or scanning forward */
```

Comment:
Yeah, I meant `true`, sorry. Just brainstorming around the `goto` bit.

Other thought is to pull out most of the function into another function (`ExecuteForwardScan` or something) and call that function from within `ScanDirectionIsForward`. This function's getting pretty long, so I might be OK with that.

I'm not terribly averse to the `goto`, but only if necessary; was just trying to think up ways where it might not be necessary that preserve readability (I tend to reserve `goto` for resource cleanup and crap that results in many places all jumping to the same "footer" label. The fact that there's only a single jump here feels like it could be avoided).